rio_tiler/io/base.py

```diff
@@ -0,0 +1,101 @@
+"""rio_tiler.io.base: ABC class for rio-tiler readers."""
+
+import abc
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Sequence, Tuple
+
+import numpy
+
+
+# ref: https://github.com/python/mypy/issues/5374
+@dataclass  # type: ignore
+class BaseReader(metaclass=abc.ABCMeta):
+    """Rio-tiler.io BaseReader."""
+
+    bounds: Sequence[float] = field(init=False)
```

Comment:
bounds will be set in `__enter__`, I'm not sure if there is a best way to tell the class